src/OmniSharp.MSBuild/ProjectFile/AnalyzerAssemblyLoader.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+
+namespace OmniSharp.MSBuild.ProjectFile
+{
+    public class AnalyzerAssemblyLoader : IAnalyzerAssemblyLoader
+    {
+        private readonly ConcurrentDictionary<string, Assembly> _assemblyPaths = new ConcurrentDictionary<string, Assembly>();
```

Comment:
If you're creating a map keyed on path, you should pass `StringComparer.OrdinalIgnoreCase`.